web-registry/src/pages/Media.tsx

```diff
@@ -1,39 +1,68 @@
 import React from 'react';
 import { useNavigate, useParams } from 'react-router-dom';
 
+import {
+  getURLInitialValue,
+  getURLListInitialValue,
+} from 'web-components/lib/utils/schemaURL';
+import { Loading } from 'web-components/lib/components/loading';
+
 import {
   OnboardingFormTemplate,
   EditFormTemplate,
 } from '../components/templates';
-import { MediaForm, MediaValues } from '../components/organisms';
+import {
+  MediaForm,
+  MediaValues,
+  isSimpleValues,
+} from '../components/organisms';
+import { useProjectEditContext } from '../pages/ProjectEdit';
+import { getProjectShapeIri } from '../lib/rdf';
 import {
   useProjectByIdQuery,
+  useShaclGraphByUriQuery,
   useUpdateProjectByIdMutation,
 } from '../generated/graphql';
-import { useProjectEditContext } from '../pages/ProjectEdit';
 
-const Media: React.FC = () => {
+const Media = (): JSX.Element => {
   const { projectId } = useParams();
   const navigate = useNavigate();
   const { isEdit } = useProjectEditContext();
-
   const [updateProject] = useUpdateProjectByIdMutation();
-  const { data } = useProjectByIdQuery({
+  const { data: projectData, loading: loadingProject } = useProjectByIdQuery({
     variables: { id: projectId },
     fetchPolicy: 'cache-and-network',
   });
-  const project = data?.projectById;
+  const project = projectData?.projectById;
   const creditClassId = project?.creditClassByCreditClassId?.onChainId;
 
-  let initialFieldValues: MediaValues | undefined;
-  if (project?.metadata) {
-    const metadata = project.metadata;
-    initialFieldValues = {
-      'regen:previewPhoto': metadata['regen:previewPhoto'],
-      'regen:galleryPhotos': metadata['regen:galleryPhotos'],
-      'regen:landStewardPhoto': metadata['regen:landStewardPhoto'],
-      'regen:videoURL': metadata['regen:videoURL'],
-    };
+  const { data: graphData } = useShaclGraphByUriQuery({
+    skip: !project,
+    variables: {
+      uri: getProjectShapeIri(creditClassId),
+    },
+  });
+
+  function getInitialFormValues(): MediaValues {
+    let values: MediaValues = {};
+    const metadata = project?.metadata || {};
+    values['regen:previewPhoto'] = getURLInitialValue(
+      metadata['regen:previewPhoto'],
+    );
+    values['regen:galleryPhotos'] = getURLListInitialValue(
+      4,
```

Comment:
Should we extract this value as const at the top of the file in order to name it ?

Reply:
sorry @flagrede - i'm not quite sure I follow. Are you suggesting extracting the `getURLListInitialValue` to a const? like
```ts
const getGalleryInitialValue = (metadata: UrlList): UrlList =>
  getURLListInitialValue(4, metadata);
```

Reply:
Sorry, it wasn't clear, I was suggesting moving `4` as a constant.
Having static numbers and string extracted as constant allows to name them (here for example it would give more information about what `4` stand for) and also having them grouped and separated make them more easily configurable if needed.